src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenStackAllocInitializerTests.cs

```diff
@@ -949,6 +1078,518 @@ .maxstack  4
+                  IL_0002:  conv.u
+                  IL_0003:  localloc
+                  IL_0005:  dup
+                  IL_0006:  ldsfld     "decimal decimal.One"
+                  IL_000b:  stobj      "decimal"
+                  IL_0010:  dup
+                  IL_0011:  ldc.i4.s   16
+                  IL_0013:  add
+                  IL_0014:  ldc.i4.2
+                  IL_0015:  newobj     "decimal..ctor(int)"
+                  IL_001a:  stobj      "decimal"
+                  IL_001f:  dup
+                  IL_0020:  ldc.i4.2
+                  IL_0021:  conv.i
+                  IL_0022:  ldc.i4.s   16
+                  IL_0024:  mul
+                  IL_0025:  add
+                  IL_0026:  ldc.i4.3
+                  IL_0027:  newobj     "decimal..ctor(int)"
+                  IL_002c:  stobj      "decimal"
+                  IL_0031:  call       "void C.Write(decimal*)"
+                  IL_0036:  ret
+                }
+                """);
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69325")]
+        public void TestMixed()
+        {
+            var source = """
+                using System;
+                static class C
+                {
+                    static void Main() => M(4);
+
+                    static void M(int i)
+                    {
+                        ReadOnlySpan<int> p = stackalloc int[] { 1, 2, 3, i };
+                        Write(p);
+                    }
+
+                    static void Write(ReadOnlySpan<int> span)
+                    {
+                        foreach (int x in span)
+                            Console.Write(x);
+                    }
+                }
+                """;
+            var verifier = CompileAndVerify(source, expectedOutput: ExecutionConditionUtil.IsCoreClr ? "1234" : null,
+                verify: Verification.Fails, targetFramework: TargetFramework.Net70);
+            verifier.VerifyDiagnostics();
+            verifier.VerifyIL("C.M", """
+                {
+                  // Code size       57 (0x39)
+                  .maxstack  5
+                  .locals init (System.ReadOnlySpan<int> V_0)
+                  IL_0000:  ldc.i4.s   16
+                  IL_0002:  conv.u
+                  IL_0003:  localloc
+                  IL_0005:  dup
+                  IL_0006:  ldtoken    "<PrivateImplementationDetails>.__StaticArrayInitTypeSize=16_Align=4 <PrivateImplementationDetails>.81C1A5A2F482E82CA2C66653482AB24E6D90944BF183C8164E8F8F8D72DB60DB4"
+                  IL_000b:  call       "System.ReadOnlySpan<int> System.Runtime.CompilerServices.RuntimeHelpers.CreateSpan<int>(System.RuntimeFieldHandle)"
+                  IL_0010:  stloc.0
+                  IL_0011:  ldloca.s   V_0
+                  IL_0013:  ldc.i4.0
+                  IL_0014:  call       "ref readonly int System.ReadOnlySpan<int>.this[int].get"
+                  IL_0019:  ldc.i4.s   16
+                  IL_001b:  unaligned. 4
```

Comment:
>IL_001b:  unaligned. 4 [](http://example.com/codeflow?start=18&length=22)

It doesn't look like there is a test scenario with alignment 2. Consider adding one.